lib/pkgr/builder.rb

```diff
@@ -284,17 +287,23 @@ def distribution
       @distribution ||= Distributions.current(config)
     end
 
-    # List of available buildpacks for the current distribution.
-    def buildpacks
-      distribution.buildpacks
-    end
-
-    # Buildpack detected for the app, if any.
-    def buildpack_for_app
+    # Buildpacks detected for the app, if any. If multiple buildpacks are explicitly specified, all are used
+    def buildpacks_for_app
       raise "#{source_dir} does not exist" unless File.directory?(source_dir)
-      @buildpack_for_app ||= buildpacks.find do |buildpack|
-        buildpack.setup(config.edge, config.home)
-        buildpack.detect(source_dir)
+      @buildpacks_for_app ||= begin
+        mode, buildpacks = distribution.buildpacks
+        case mode
+        when :custom
+          buildpacks.find_all do |buildpack|
+            buildpack.setup(config.edge, config.home)
+            buildpack.detect(source_dir)
+          end
+        else
+          [buildpacks.find do |buildpack|
+            buildpack.setup(config.edge, config.home)
+            buildpack.detect(source_dir)
+          end].compact
```

Comment:
In the current implementation, an order favorites ruby buildpack. e.g  from centos-7
```
https://github.com/pkgr/heroku-buildpack-ruby.git#universal,BUILDPACK_NODE_VERSION="0.6.8",CURL_CONNECT_TIMEOUT=60,CURL_TIMEOUT=300
https://github.com/heroku/heroku-buildpack-nodejs.git#v58
https://github.com/kr/heroku-buildpack-go.git
```
Buildpack check is quite primitive `if Gemfile then ruby`, `if package.json then node`, when I have both in the app directory ruby wins :) I was quite surprised at first glance.

If we want to be precise we should explicitly define only one buildpack.
Maybe multi buildpacks as a default is not a bad solution cc @crohr